### PR TITLE
Add support for FunctionTemplate signature checking

### PR DESCRIPTION
**Pull request by galpeter**
The FunctionTemplate's signature is used to check if a method
is invoked on a correct instance.

### CODE REVIEW

File: deps/jerry-v8/tests/function_template.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        // Without a signature a template method it should work
+        {
+            v8::Local<v8::FunctionTemplate> getprop_method = v8::FunctionTemplate::New(env.getIsolate(), method_demo);
+            prototype->SetAccessorProperty(getPropOkString, getprop_method, v8::Local<v8::FunctionTemplate>());
+        }
+    }
+
+    v8::Local<v8::Function> func = tmplt->GetFunction();
+    {
+      env.getContext()->Global()->Set(
```
Comment (rtakacs) [inline deps/jerry-v8/tests/function_template.cpp]: Wrong indentation

File: deps/jerry-v8/tests/function_template.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        // By configuring the signature for a template method it should throw an error if an incorrect "this" value is used during access.
+        {
+            v8::Local<v8::Signature> signature = v8::Signature::New(env.getIsolate(), tmplt);
+            v8::Local<v8::FunctionTemplate> getprop_method =
+                v8::FunctionTemplate::New(env.getIsolate(), method_demo, v8::Local<v8::External>(), signature);
+
+            prototype->SetAccessorProperty(getPropErrorString, getprop_method, v8::Local<v8::FunctionTemplate>());
+        }
+
+        // Without a signature a template method it should work
```
Comment (rtakacs) [inline deps/jerry-v8/tests/function_template.cpp]: "it" seems unnecessary

**Comment by galpeter** [comment]
Fixed & updated the PR

**Review by rtakacs — APPROVED** [review]
LGTM

**Pull request merged**